server/tserver/src/main/java/org/apache/accumulo/tserver/memory/LargestFirstMemoryManager.java

```diff
@@ -60,8 +57,7 @@ public class LargestFirstMemoryManager {
   private double compactionThreshold;
   private long maxObserved;
   private final HashMap<TableId,Long> mincIdleThresholds = new HashMap<>();
-  private final Cache<TableId,Long> mincAgeThresholds =
-      Caffeine.newBuilder().expireAfterWrite(5, TimeUnit.MINUTES).build();
+  private final HashMap<TableId,Long> mincAgeThresholds = new HashMap<>();
```

Comment:
These maps are only used in a single method.  Ideally they would be scoped to that method instead of the class, however the code is structed in a more complex way so that test can override behavior.

Reply:
I looked into this a little more and now I'm thinking that the additional layer of caching that we have in the codebase on top of the configuration should be removed. I think it's making the code **less** responsive to configuration changes. 

Example:
```
    return mincIdleThresholds.computeIfAbsent(tableId, tid -> context.getTableConfiguration(tid)
        .getTimeInMillis(Property.TABLE_MINC_COMPACT_IDLETIME));
```

If you look into `ServerContext.getTableConfiguration(TableId)` it calls `ServerConfigurationFactory.getTableConfiguration`, which puts the resulting `TableConfiguration` into a `Cache` that expires after 1 hour **and** it registers a listener on the underlying `ZooPropStore` such that it clears the `TableConfiguration` entry in the `Cache` when a change occurs. The `ZooPropStore` has a `Watcher` that fires when things change in `ZooKeeper`.

Thoughts?

Reply:
Removing this bit of caching would simplify this code. The map in this case is very short lived, so I don't think it adds latency in this case however it does add complexity to the code that makes it harder to maintain.  Looking at the way this code works the main benefit of this little map is that it avoids reparsing the duration for each tablet of the same table.  Without the map we would keep converting at string like `5m` to milliseconds over and over as we examine all the tablets in the tablet server.  However instead of doing that outside of the configuration in some places we should probably push that down into the configuration and cache the result of parsing `5m` inside the configuration objs until the configuration changes.  If we cached the parsing inside the cache that could make #4742 more performant where it caches Duration objects and does not create them on each call.

For this PR we could just remove the map its using to avoid reparsing and open an issue about caching duration parsing inside the cache.

Reply:
Opened #4860.  For this PR I will look into removing the map.